butte-build/src/codegen.rs

```diff
@@ -339,37 +411,77 @@ impl ToTokens for ir::Table<'_> {
-                            Some(#enum_ident::#variant_ident) => self.table
-                                .get::<#variant_ty_wrapped>(#struct_id::#offset_name)?
-                                .map(#union_ident::#variant_ident)
+                if field.metadata.required {
+                    let names_to_enum_variant = variants.iter().map(
+                        |ir::UnionVariant {
+                             ident: variant_ident,
+                             ty: variant_ty,
+                         }| {
+                            let variant_ty_wrapped = to_type_token(
+                                variant_ty,
+                                &quote!('a),
+                                &quote!(butte::ForwardsUOffset),
+                                true,
+                            );
+                            quote! {
+                                #enum_ident::#variant_ident => #union_ident::#variant_ident(self.table
+                                    .get::<#variant_ty_wrapped>(#struct_id::#offset_name)?
+                                    .ok_or_else(|| butte::Error::RequiredFieldMissing(#snake_name_str))?)
+                            }
+                        },
+                    );
+                    quote! {
+                        #[inline]
+                        pub fn #snake_name<'a>(&'a self) -> Result<#ty_ret, butte::Error> {
+                            Ok(match self.#type_snake_name()? {
+                              #(#names_to_enum_variant),*,
+                              #enum_ident::None => return Err(butte::Error::RequiredFieldMissing(#snake_name_str))
+                            })
                         }
-                    },
-                );
+                    }
+                } else {
+                    let names_to_enum_variant = variants.iter().map(
+                        |ir::UnionVariant {
+                             ident: variant_ident,
+                             ty: variant_ty,
+                         }| {
+                            let variant_ty_wrapped = to_type_token(
+                                variant_ty,
+                                &quote!('a),
+                                &quote!(butte::ForwardsUOffset),
+                                true,
+                            );
+                            quote! {
+                                Some(#enum_ident::#variant_ident) => self.table
+                                    .get::<#variant_ty_wrapped>(#struct_id::#offset_name)?
+                                    .map(#union_ident::#variant_ident)
+                            }
+                        },
+                    );
+
+                    quote! {
+                        #[inline]
+                        pub fn #snake_name<'a>(&'a self) -> Result<Option<#ty_ret>, butte::Error> {
+                            Ok(match self.#type_snake_name()? {
+                              #(#names_to_enum_variant),*,
```

Comment:
nit: similar comment to the above about two commas.

Reply:
Fixed